### PR TITLE
Fix bug: exception while creating audit table schema when using single table inheritance

### LINKED ISSUE #132: exception while creating audit table schema when using single table inheritance
| Q                    | A
| ---------------------| -----------------------
| `auditor` version    | 2.3.0
| PHP version          | 8.1.8
| Database             | MySQL / SQLite

#### Summary

I have an abstract parent entity which is not audited, but uses doctrine/orm inheritance type "SINGLE_TABLE" and defines the database table for its child entities. Some child entities are configured to be audited.

With auditor 2.3.0, creating the audit database schema throws an exception.

#### Current behavior

When the database schema is created for this abstract parent entity, the CreateSchemaListener wants to create the corresponding audit table. This throws an exception while trying to determine the table name from the configuration with the new logic introduced in auditor 2.3.0, as the parent entity does not have any configuration: `Undefined array key "App\AbstractParentEntity"`

https://github.com/DamienHarper/auditor/blob/19ff16a658e7bf268dbea7b88c08118da5caf1ae/src/Provider/Doctrine/Persistence/Schema/SchemaManager.php#L173

In versions before 2.3.0, the table name from the event was being used, which is independent of the configuration.

#### How to reproduce

Define an abstract parent entity and an audited child entity:

```
namespace App;

/**
 * @ORM\Table(name="entities")
 * @ORM\Entity()
 * @ORM\InheritanceType("SINGLE_TABLE")
 */
abstract class AbstractParentEntity {}

/**
 * @ORM\Entity
 */
class Entity {}
```

Configure auditing for the child entity:

```
dh_auditor:
    enabled: true
    providers:
        doctrine:
            table_prefix: ~
            table_suffix: '_audit'
            entities:
                App\Entity: ~
```

Create the database schema using `Doctrine\ORM\Tools\SchemaTool::createSchema` for the parent entity to create an `Doctrine\ORM\Tools\ToolEvents::postGenerateSchemaTable` event.

#### Expected behavior

Creating the database schema for the abstract parent entity and its child entities does not throw an exception and creates the corresponding audit table schema.

### PR DESCRIPTION
**Pull request by dmitryuk**
Closes #132

### CODE REVIEW

**Comment by dmitryuk** [comment]
@DamienHarper please watch, because I have the same issue in latest release.

File: src/Provider/Doctrine/Configuration.php [hunk 1]
```diff
@@ -49,6 +37,8 @@ class Configuration implements ConfigurationInterface
 
     private array $annotationLoaded = [];
 
+    private bool $inited = false;
```
Comment (DamienHarper) [inline src/Provider/Doctrine/Configuration.php:40]: Typo? I'd name it `$initialized`

File: src/Provider/Doctrine/Configuration.php [hunk 2]
```diff
@@ -30,16 +28,6 @@ class Configuration implements ConfigurationInterface
...
-    /**
-     * @var array<StorageServiceInterface>
-     */
-    private array $storageServices = [];
-
-    /**
-     * @var array<AuditingServiceInterface>
-     */
-    private array $auditingServices = [];
-
```
Comment (DamienHarper) [inline src/Provider/Doctrine/Configuration.php:42]: This should also be removed from `Configuration::configureOptions()`

**Review by DamienHarper — CHANGES_REQUESTED** [review]
Thanks again @dmitryuk for this PR. 

I started working on version `3.0.0` in `master` branch a few days ago (it will drop PHP 7.4 support, Symfony 4.x support etc.)

I also created a `2.x` branch since I started working on `3.0` so, could you target this PR to `2.x` branch please?

I'll backport the fix to `master` branch once merged and released as `2.3.1`

File: src/Provider/Doctrine/Configuration.php (hunk 2 above)
Comment (dmitryuk) [reply src/Provider/Doctrine/Configuration.php]: done

File: src/Provider/Doctrine/Configuration.php (hunk 1 above)
Comment (dmitryuk) [reply src/Provider/Doctrine/Configuration.php]: done, thanks

**Comment by dmitryuk** [comment]
> Thanks again @dmitryuk for this PR.
> 
> I started working on version `3.0.0` in `master` branch a few days ago (it will drop PHP 7.4 support, Symfony 4.x support etc.)
> 
> I also created a `2.x` branch since I started working on `3.0` so, could you target this PR to `2.x` branch please?
> 
> I'll backport the fix to `master` branch once merged and released as `2.3.1`

Done

**Comment by dmitryuk** [comment]
Don't understand why Github workflow not run after push

**Review by DamienHarper — APPROVED** [review]

**Comment by DamienHarper** [comment]
> Don't understand why Github workflow not run after push

Fixed now but CI is failing now

**Pull request merged**